jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -127,6 +127,9 @@ public HttpCookie(String name, String value, String domain, String path, long ma
         _version = version;
         _expiration = maxAge < 0 ? -1 : NanoTime.now() + TimeUnit.SECONDS.toNanos(maxAge);
         _attributes = (attributes == null ? Collections.emptyMap() : attributes);
+
+        // remove attributes that conflict with other values here
+        List.of("Domain", "Path", "Max-Age", "HttpOnly", "Secure", "Comment").forEach(_attributes::remove);
```

Comment:
make this list final static field rather than recreate in every constructor (Java might do that already... but let's not rely on it)

Reply:
Note also that this is mutating the passed in map, which is not exactly best practice.  At least this should be javadoc'd.  Maybe we create new map if we need to mutate it? maybe we always create a new map?

Reply:
It needs improvement in other ways as well.

```
2023-01-18 21:41:57.389:WARN :oejes.ServletChannel:qtp1687702287-501: /another.dump
java.lang.UnsupportedOperationException
	at java.base/java.util.Collections$UnmodifiableMap.remove(Collections.java:1508)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at org.eclipse.jetty.http.HttpCookie.<init>(HttpCookie.java:132)
	at org.eclipse.jetty.session.Session.generateSetCookie(Session.java:156)
	at org.eclipse.jetty.ee10.servlet.SessionHandler.getSessionCookie(SessionHandler.java:441)
	at org.eclipse.jetty.ee10.servlet.ServletApiRequest.getSession(ServletApiRequest.java:424)
	at org.eclipse.jetty.ee10.demos.DumpServlet.doGet(DumpServlet.java:44)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:527)
```

Reply:
We should remove them, otherwise they get added to the generated cookie twice.

Reply:
Pushed a change to this, including the requested javadoc.